gobblin-utility/src/main/java/gobblin/util/HadoopUtils.java

```diff
@@ -63,11 +64,17 @@ public static Configuration newConfiguration() {
     return results;
   }
 
+  public static List<FileStatus> listStatusRecursive(FileSystem fileSystem, Iterable<Path> paths) throws IOException {
```

Comment:
`ListStatusRecursive` in `HadoopUtils` is overlapped with `listFilesRecursively` in `FileListUtils`. Can you take a look at `FileListUtils` and add this method there?